redbot/cogs/streams/streamtypes.py

```diff
@@ -280,22 +282,67 @@ def __init__(self, **kwargs):
         self.id = kwargs.pop("id", None)
         self._client_id = kwargs.pop("token", None)
         self._bearer = kwargs.pop("bearer", None)
+        self._rate_limit_resets: set = set()
+        self._rate_limit_remaining: int = 0
         super().__init__(**kwargs)
 
+    async def wait_for_rate_limit_reset(self) -> None:
+        """Check rate limits in response header and ensure we're following them.
+
+        From python-twitch-client and adaptated to asyncio from Trusty-cogs:
+        https://github.com/tsifrer/python-twitch-client/blob/master/twitch/helix/base.py
+        https://github.com/TrustyJAID/Trusty-cogs/blob/master/twitch/twitch_api.py
+        """
+        if self._rate_limit_remaining == 0:
+            current_time = int(time.time())
+            self._rate_limit_resets = set(x for x in self._rate_limit_resets if x > current_time)
+
+            if self._rate_limit_resets:
+                reset_time = next(self._rate_limit_resets)
```

Comment:
turns out this needs to be wrapped in `iter()` :)
```suggestion
                reset_time = next(iter(self._rate_limit_resets))
```